core/src/main/scala/org/http4s/syntax/StringSyntax.scala

```diff
@@ -11,4 +11,5 @@ trait StringSyntax {
 final class StringOps(val self: String) extends AnyVal {
   def ci: CaseInsensitiveString =
     CaseInsensitiveString(self)
+  def path: Uri.Path = Uri.Path.fromString(self)
```

Comment:
We've been drifting toward StringContexts for most things.  Maybe better to have `path"/foo"` instead?